### PR TITLE
Angular comment design

### LINKED ISSUE #141: Comment Material Design
User stories:

- #6

### PR DESCRIPTION
**Pull request by benedekhalaj**
- Closes #141

### CODE REVIEW

File: angular/src/app/components/comment/comment.component.html [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+  <mat-card>
+    <mat-card-content>
+      <div mat-card-avatar class="comment-image"></div>
+      <div class="comment-details">
+        <strong><a class="comment-user-link" routerLink="/profile/{{comment.user.id}}">{{comment.user.username}}</a></strong>
+        <app-date-ago class="comment-timestamp" [date]="comment.timestamp"></app-date-ago>
+        <p class="comment-description">{{comment.description}}</p>
+      </div>
+    </mat-card-content>
+  </mat-card>
```
Comment (dudaskobendeguz) [inline angular/src/app/components/comment/comment.component.html:11]: clever and minimalist, i like it :dagger:

File: angular/src/app/components/comment/comment.component.ts [hunk 2]
```diff
@@ -9,8 +9,14 @@ import {PostComment} from "../../interfaces/post-comment";
...
   @Input() comment?: PostComment
+  date?: string
 
   constructor() { }
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    if (this.comment) {
+      const date = new Date(this.comment.timestamp);
+      this.date = `${date.getFullYear()}-${date.getMonth()}-${date.getDate()} ${date.getHours()}:${date.getUTCMinutes()}`
```
Comment (dudaskobendeguz) [inline angular/src/app/components/comment/comment.component.ts]: maybe you have to delete this "useless code" :kaaba:

File: angular/src/app/components/comment/comment.component.ts [hunk 3]
```diff
@@ -9,8 +9,14 @@ import {PostComment} from "../../interfaces/post-comment";
 export class CommentComponent implements OnInit {
 
   @Input() comment?: PostComment
+  date?: string
```
Comment (dudaskobendeguz) [inline angular/src/app/components/comment/comment.component.ts]: delete this one line too(don't forget to remove this form html as well) :cactus:

File: angular/src/app/components/user-post-components/user-post-detail/user-post-detail.component.html [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
     <app-user-post
       [userPost]="userPost"
       (onDeleteUserPost)="deleteUserPost($event)"
-      (onUpdateUserPost)="updateUserPost($event)"></app-user-post>
-    <button mat-flat-button type="button" color="primary" (click)="openAddCommentDialog()">Add Comment</button>
-    <div class="user-post-comments" *ngFor="let comment of userPost.comments">
-      <app-comment [comment]="comment"></app-comment>
-    </div>
+      (onUpdateUserPost)="updateUserPost($event)"
+    ></app-user-post>
```
Comment (dudaskobendeguz) [inline angular/src/app/components/user-post-components/user-post-detail/user-post-detail.component.html:9]: nice refactor :+1:

**Review by dudaskobendeguz — CHANGES_REQUESTED** [review]
well done :+1:  :sake:

File: angular/src/app/components/comment/comment.component.ts (hunk 2 above)
Comment (benedekhalaj) [reply angular/src/app/components/comment/comment.component.ts]: thanks for your polite suggestion

**Review by dudaskobendeguz — APPROVED** [review]

**Pull request merged**